### ISSUE TITLE
Rna-Seq Pipeline, aligning with STAR but still running bowtie for ht-seq output 

### ISSUE THREAD

**Issue by kuikuisven** [chapmanb/bcbio-nextgen#1099]
Hi Brad, 

This may be a naive question, but i have been running the pipeline with star to align some single cell RNA seq and hoped that the whole analysis would be fast. Now the alignment was indeed very fast, and i discovered that (at least) for single cell RNA-Seq, it's better to actually use one instance of STAR sequentially on all the samples, rather than having several in parallel. Now the pipeline is computing read counts, and it's running bowtie, disregarding the output of star ... is there something i'm doing wrong here?
also, is there a single cell RNA-Seq pipeline in preparation? 

best,

Nico

**Comment by roryk** [comment]
Hi Nico,

Thanks for the note. I think bowtie running isn't expected if you are aligning with STAR-- in which step is bowtie running? There shouldn't be anything different you have to do for single-cell RNAseq for the steps that bcbio-nextgen handles, just running it normally will work fine.

You can up the number of cores STAR uses by editing the bcbio_system_yaml.file to make it run a bit faster. If you set the number of cores to be the number of cores on the machine you are using, it will run STAR with that many cores. Since STAR is memory intensive usually it gets limited by the amount of memory on the node rather than the number of cores, though.

**Comment by kuikuisven** [comment]
oh, that was fast— the pipeline is populating the htseq-count folder now, and this is the step where it’s running bowtie again…

about STAR, i ll keep this in mind. but for what i saw, each run of STAR  starts by putting the genome into memory, which for my case was the limiting factor, because of slow I/O. I know there’s a STAR option to make the genome stay in memory between runs. is there a way to pass a flag to STAR from the resource section in the project configuration yaml file?


On 06 Nov 2015, at 15:05, Rory Kirchner <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Nico,

Thanks for the note. I think bowtie running isn't expected if you are aligning with STAR-- in which step is bowtie running? There shouldn't be anything different you have to do for single-cell RNAseq for the steps that bcbio-nextgen handles, just running it normally will work fine.

You can up the number of cores STAR uses by editing the bcbio_system_yaml.file to make it run a bit faster. If you set the number of cores to be the number of cores on the machine you are using, it will run STAR with that many cores. Since STAR is memory intensive usually it gets limited by the amount of memory on the node rather than the number of cores, though.

**Comment by roryk** [comment]
Hi Nico,

Could you upload your bcbio-nextgen-debug.log file where it is running bowtie so we can take a look? It shouldn't be running that for the counting step.

We don't use the shared memory feature of STAR because we're a little worried about being a nightmare to debug on a cluster and lots of peoples configurations won't support it out of the box. I agree for single-cell where you have a low number of counts it would be better to not load the STAR index into memory repeatedly though.

**Comment by roryk** [comment]
gist.github.com is good place. Thanks!

**Comment by kuikuisven** [comment]
Here:
<script src="https://gist.github.com/kuikuisven/6a01d58c5d0a089362c9.js"></script>

**Comment by roryk** [comment]
Thanks, that was super helpful. There are some versions of STAR that segfault when generating transcriptome-only alignments, so we had a check in there to handle those versions. I broke the logic of the check earlier which was causing this to always skip doing the transcriptome-only alignments. I fixed that and it should now not use the fallback mechanism when running STAR.

For your project rather than restarting I'd just let bowtie2 finish, it shouldn't take a huge amount of time since it is just aligning to the transcriptome. In the future though STAR will handle this itself and be much faster. Thanks for helping us debug!
